lib/matplotlib/tests/test_polar.py

```diff
@@ -506,3 +506,12 @@ def test_polar_errorbar(order):
         ax.errorbar(theta, r, xerr=0.1, yerr=0.1, capsize=7, fmt="o", c="seagreen")
         ax.set_theta_zero_location("N")
         ax.set_theta_direction(-1)
+
+
+def test_radial_limits_behavior():
+    fig = plt.figure()
+    ax = fig.add_subplot(projection='polar')
+    ax.set_rticks([1, 2, 3, 4])
+    assert ax.get_ylim()[0] == 0
+    ax.plot([0, 1], [2, 3])
+    assert ax.get_ylim()[0] == 0
```

Comment:
```suggestion
I'd more explicitly write out what is tested, and also include the case of negative ticks/data.

```suggestion
    # r=0 is kept as limit if positive data and ticks are used
    # negative ticks or data result in negativ limits
    fig = plt.figure()
    ax = fig.add_subplot(projection='polar')
    assert ax.get_ylim() == (0, 1)
    # upper limit is expanded to include the ticks, but lower limit stays at 0
    ax.set_rticks([1, 2, 3, 4])
    assert ax.get_ylim() == (0, 4)
    # upper limit is autoscaled to data, but lower limit limit stays 0
    ax.plot([1, 2], [1, 2])
    assert ax.get_ylim() == (0, 2.05)
    # negative ticks also expand the negative limit
    ax.set_rticks([-1, 0, 1, 2])
    assert ax.get_ylim() == (-1, 2.05)
    # negative data also autoscales to negative limits
    ax.plot([1, 2], [-1, -2])
    ax.get_ylim() == (-2.2, 2.2)
``````

Reply:
Sorry for the oversight, I was more focused on testing the example from the reported Bug and assumed the rest of the behaviour was already tested elsewhere. Thanks for the correction